Release session while waiting for batch signal

jraddaoui opened the pull request:
Complete the current session before waiting for the batch signal to allow other sessions to execute in the worker. Recreate the session after in the same worker to continue having access to the local files.

The recreated session takes an activity slot in the same worker where the functional activities run. Set `MaxConcurrentActivityExecutionSize` to the double of `MaxConcurrentSessionExecutionSize` in both workers to avoid deadlocks in this situation.

Comes from #1456. Based in #1460. Refs #1405.

jraddaoui commented on internal/workflow/processing.go:
```diff
@@ -205,20 +205,20 @@ func (w *ProcessingWorkflow) Execute(ctx temporalsdk_workflow.Context, req *inge
 		maxAttempts := 5
 
-		activityOpts := temporalsdk_workflow.WithActivityOptions(ctx, temporalsdk_workflow.ActivityOptions{
+		ctx = temporalsdk_workflow.WithActivityOptions(ctx, temporalsdk_workflow.ActivityOptions{
```
https://github.com/artefactual-sdps/enduro/pull/1456#discussion_r2607996005

jraddaoui commented on internal/workflow/processing.go:
```diff
@@ -287,12 +287,10 @@ func (w *ProcessingWorkflow) Execute(ctx temporalsdk_workflow.Context, req *inge
 ) error {
 	// Cleanup session files on exit.
-	defer w.sessionCleanup(sessCtx, state)
+	defer func() { w.sessionCleanup(sessCtx, state) }()
```
https://github.com/artefactual-sdps/enduro/pull/1456#discussion_r2607997336

jraddaoui reviewed (commented):
@djjuhasz, check https://github.com/artefactual-sdps/enduro/pull/1456#discussion_r2602287081 for a bit more info about the concurrent activities change. About moving the contexts around, I thought about adding them to the state, but it feels odd to hide the context that way.

djjuhasz reviewed (approved):
Wow, good work @jraddaoui.  The batch processing signalling is a lot more complicated than I anticipated when we started discussing handling batches and this is a clever way to keep processing on the same worker when all the SIPs validate without having to keep the session open the whole time.

jraddaoui committed:
Release session while waiting for batch signal

Complete the current session before waiting for the batch signal to
allow other sessions to execute in the worker. Recreate the session
after in the same worker to continue having access to the local files.

The recreated session takes an activity slot in the same worker where
the functional activities run. Set `MaxConcurrentActivityExecutionSize`
to the double of `MaxConcurrentSessionExecutionSize` in both workers
to avoid deadlocks in this situation.

[skip-codecov]

jraddaoui commented:
Thanks @djjuhasz!
